packages/google_sign_in/google_sign_in/lib/google_sign_in.dart

```diff
@@ -296,11 +296,10 @@ class GoogleSignIn {
 
   // Performs initialization, guarding it with the _initialization future.
   Future<void> _ensureInitialized() async {
-    return _initialization ??= _doInitialization()
-      ..catchError((dynamic _) {
-        // Invalidate initialization if it errors out.
-        _initialization = null;
-      });
+    return (_initialization ??= _doInitialization()).catchError((dynamic _) {
+      // Invalidate initialization if it errors out.
+      _initialization = null;
+    });
```

Comment:
Apparently the fact that the error wasn't being caught by the returned future was (or at least has become) intentional behavior, since there was a unit test that this change broke. I doubt very much that having `_initialization` not be guaranteed to be null when the future completes is intentional though, so I changed it to return the `catchError` wrapper, but made the wrapper rethrow after nulling `_initialization`.